container_pipeline/workers/base.py

```diff
@@ -0,0 +1,65 @@
+import json
+import logging
+import os
+
+from container_pipeline.lib import settings
+from container_pipeline.lib.queue import JobQueue
+from container_pipeline.lib.log import DynamicFileHandler
+
+
+class BaseWorker(object):
+    """Base class for pipeline workers"""
+
+    def __init__(self, logger=None, sub=None, pub=None):
+        self.logger = logger or logging.getLogger('console')
+        self.queue = JobQueue(host=settings.BEANSTALKD_HOST,
+                              port=settings.BEANSTALKD_PORT,
+                              sub=sub, pub=pub, logger=self.logger)
+
+    def handle_job(self, job):
+        """
+        This method is called to process job data from task queue.
+        """
+        raise NotImplementedError
+
+    def notify(self, data):
+        """
+        This method queues user notifications to be processed by the
+        mail service worker. Customize as needed.
+        """
+        self.queue.put(json.dumps(data), 'master_tube')
+
+    def export_logs(self, logs, destination):
+        """"Write logs in given destination"""
+        self.logger.debug('Writing build logs to NFS share')
```

Comment:
Keeping the debug message argument optional with a default (Generic) message in method itself.

@rtnpro : WDYT ?